pkg/network/usm/utils/file_registry.go

```diff
@@ -198,6 +190,17 @@ func (r *FileRegistry) Register(namespacedPath string, pid uint32, activationCB,
 	return nil
 }
 
+// addToBlocklist adds the pathID and matching sample path to the blocklist.
+func (r *FileRegistry) addToBlocklist(pathID PathIdentifier, hostPath string) {
+	for _, entry := range r.blocklistByID.Keys() {
+		if entry.PathIdentifier == pathID {
+			return
+		}
+	}
+
+	r.blocklistByID.Add(PathIdentifierWithSamplePath{pathID, hostPath}, struct{}{})
+}
```

Comment:
I missed that in the original review, but, the PR can be dramatically simplified, by changing 
`blocklistByID *simplelru.LRU[PathIdentifierWithSamplePath, struct{}]`
to
`blocklistByID *simplelru.LRU[PathIdentifier, string]`